File: LeetCode/Easy/1260-shift-2d-grid/1260-shift-2d-grid.java
import java.util.*;

class Solution {
    public List<List<Integer>> shiftGrid(int[][] grid, int k) {
        int m = grid.length, n = grid[0].length, total = m * n;
        Integer[][] ans = new Integer[m][n];
        k %= total;
        for (int i = 0; i < total; i++) {
            int next = (i + k) % total;
            ans[next / n][next % n] = grid[i / n][i % n];
        }
        List<List<Integer>> result = new ArrayList<>();
        for (Integer[] row : ans) result.add(Arrays.asList(row));
        return result;
    }
}

File: LeetCode/Easy/1260-shift-2d-grid/1260-shift-2d-grid.cpp
class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        int m = grid.size();
        int n = grid[0].size();
        int total = m * n;
        k %= total;
        vector<vector<int>> ans(m, vector<int>(n));
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                int oldidx = i * n + j;
                int newidx = (oldidx + k) % total;
                int nr = newidx / n;
                int nc = newidx % n;
                ans[nr][nc] = grid[i][j];
            }
        }
        return ans;
    }
};